src/OneDriveSdk.WinStore/Authentication/OnlineIdAuthenticationProvider.cs

```diff
@@ -37,6 +39,26 @@ public OnlineIdAuthenticationProvider(ServiceInfo serviceInfo)
             this.authenticator = new OnlineIdAuthenticator();
         }
 
+        public override async Task AppendAuthHeaderAsync(HttpRequestMessage request)
+        {
+            // Reauthenticate only when the token expires.
+            if (this.CurrentAccountSession != null
+                && !string.IsNullOrEmpty(this.CurrentAccountSession.AccessToken)
+                && CurrentAccountSession.ExpiresOnUtc != DateTimeOffset.MinValue
```

Comment:
Unfortunately OnlineIdAuthenticationProvider does not return an expiration on successful authentication meaning the auth token will never be refreshed here if it expires.